# Copyright © Sébastien Luttringer

# This patch is free software: you can redistribute it and/or modify
# it under the terms of the GNU Lesser General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
#
# This patch is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU Lesser General Public License for more details.
#
# You should have received a copy of the GNU Lesser General Public License
# along with this patch. If not, see <http://www.gnu.org/licenses/>.

# This patch avoid menu in Host "Edit Config" to swap every load by using
# alphabetic order.

--- a/lib/BackupPC/CGI/EditConfig.pm	2013-04-14 22:43:32.000000000 +0200
+++ b/lib/BackupPC/CGI/EditConfig.pm	2013-08-02 03:58:38.796000000 +0200
@@ -492,7 +492,7 @@
     }
 
     my $groupText;
-    foreach my $m ( keys(%ConfigMenu) ) {
+    foreach my $m ( sort( keys(%ConfigMenu) ) ) {
         next if ( $menuDisable{$m}{top} );
 	my $text = eval("qq($Lang->{$ConfigMenu{$m}{text}})");
         if ( $m eq $menu ) {
